cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -1256,14 +1256,19 @@ endif()
 # - Gandiva has a compile-time (header-only) dependency on Boost, not runtime.
 # - Tests need Boost at runtime.
 # - S3FS and Flight benchmarks need Boost at runtime.
+# - arrow_testing uses boost::filesystem. So arrow_testing requires
```

Comment:
Can we replace `boost::filesystem` with `std::filesystem`?

Reply:
No... It's used by `boost::process`...: https://github.com/apache/arrow/blob/1567be0d74c5ee90c3d989f6fcefb32c402a736c/cpp/src/arrow/testing/process.cc#L76-L92

Reply:
Ah, wait. We may be able to replace `boost::filesystem` with `std::filesystem`. We don't need to use `boost::filesystem` by replacing `boost::filesystem` with `std::filesystem` here: https://github.com/apache/arrow/blob/1567be0d74c5ee90c3d989f6fcefb32c402a736c/cpp/src/arrow/testing/process.cc#L251-L283

Reply:
Hmm. We may need to use `boost::filesystem` here...: https://github.com/apache/arrow/blob/1567be0d74c5ee90c3d989f6fcefb32c402a736c/cpp/src/arrow/testing/process.cc#L290-L292

Reply:
Fine... It seems requiring non-trivial work to replace boost::process.

Reply:
is there an issue opened for this work? I think it's worth having the issue to track reducing our boost dependencies.

Reply:
I didn't find any issue after a quick search. Is this something that worth doing? Perhaps we need to vendor a lightweight platform-independent process library like https://gitlab.com/eidheim/tiny-process-library to get rid of boost::process.

Reply:
I thought I heard @pitrou suggesting we wanted to remove our boost dependencies (or something related) but I might be mistaken

Reply:
I believe this is worthwhile and I just did similar things for the Apache Avro C++ library: https://issues.apache.org/jira/browse/AVRO-4095. However, I don't know if there is any good alternative to boost::process. If the solution is clear, I'm happy to fix this.

Reply:
Yes, unfortunately we must rely on `boost::process` for now. The only alternative AFAIK would be to write our own cross-platform process-launching code.

Reply:
In general, I agree with removing Boost dependency. But it's not easy and it requires many changes.

Let's discuss it in #45440.

BTW, can we merge this to fix test-build-cpp-fuzz failure?